### ISSUE TITLE
Unable to create sample app 

### ISSUE THREAD

**Issue by Arunkarthicknallasami** [flutter/flutter#4916]
Hi,
 I am getting following error. Help me in resolving it. I'm new to flutter

> `myapp$ flutter run
> Building APK in debug mode (android-arm)...              
> Failed to run the Flutter compiler. Exit code: 1
> /home/next/flutter/bin/cache/artifacts/engine/linux-x64/sky_snapshot: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version`GLIBCXX_3.4.18' not found (required by /home/next/flutter/bin/cache/artifacts/engine/linux-x64/sky_snapshot)`

**Comment by abarth** [comment]
The version of glibc on your linux machine is too old to run flutter's host tools.  You might need to install a newer version.  See https://github.com/flutter/flutter/blob/master/.travis.yml#L4

**Comment by eseidelGoogle** [comment]
I doubt this is very common, but if it is we could check for this with `flutter doctor` presumably?

**Comment by eseidelGoogle** [comment]
http://www.linuxquestions.org/questions/linux-software-2/how-to-check-glibc-version-263103/
My favorite part of that is that on some systems libc.so is executable. O_o

**Comment by devoncarew** [comment]
I don't have `/lib/libc.so.6` on my linux system, but it looks like we can compile and run that test app in ~45ms. So that might be a test we can reasonably do in `doctor`.

**Comment by danrubel** [comment]
We can test sky_snapshot on Linux with something equivalent to this...
```
ldd -d bin/cache/artifacts/engine/linux-x64/sky_snapshot | grep 'not found'
```

**Comment by cbracken** [comment]
Closing as dupe of #6207 (though technically this was opened first) fixed by #8798.
